### PR TITLE
Fragment Design

### PR DESCRIPTION
**Pull request by kkong006**
Change Log:
- Add Home, Like, Setting fragments to nav drawer.
- Add tab layout to Home fragment to swipe between Channel and Browse fragments.
- Add Player fragment to host mp3 player that stays at the bottom of Home activity.

### CODE REVIEW

File: app/src/main/java/appricottsoftware/clarity/HomeActivity.java [hunk 1]
```diff
@@ -73,14 +85,31 @@ public void onBackPressed() {
         moveTaskToBack(true);
     }
 
+    private void setUpPlayer() {
+        FragmentManager fragmentManager = getSupportFragmentManager();
+        fragmentManager.beginTransaction()
+                .replace(R.id.fl_home_activity_player, new PlayerFragment())
```
Comment (lepatrick714) [inline app/src/main/java/appricottsoftware/clarity/HomeActivity.java:91]: Make sure to make the beginning "." aligned with the others. 

Therefore it should be

fragmentManager.beginTransaction() 
                              .replace(....)

File: app/src/main/java/appricottsoftware/clarity/HomeActivity.java [hunk 2]
```diff
@@ -73,14 +85,31 @@ public void onBackPressed() {
...
+        homeFragment = new HomeFragment();
+        likeFragment = new LikeFragment();
+        settingFragment = new SettingFragment();
+
+        nvDrawer.setCheckedItem(R.id.nav_home_fragment);
+        FragmentManager fragmentManager = getSupportFragmentManager();
+        fragmentManager.beginTransaction()
+                .replace(R.id.fl_home_activity_main, homeFragment)
+                .commit();
+
```
Comment (lepatrick714) [inline app/src/main/java/appricottsoftware/clarity/HomeActivity.java:111]: Again here

File: app/src/main/java/appricottsoftware/clarity/HomeActivity.java [hunk 3]
```diff
@@ -107,16 +138,22 @@ private void selectDrawer(MenuItem item) {
...
-        setTitle(item.getTitle());
-        // Close nav drawer
-        drawerLayout.closeDrawers();
+        try {
+            // Insert fragment into frame layout
+            FragmentManager fragmentManager = getSupportFragmentManager();
+            fragmentManager.beginTransaction()
+                    .replace(R.id.fl_home_activity_main, fragment)
+                    .commit();
+
```
Comment (lepatrick714) [inline app/src/main/java/appricottsoftware/clarity/HomeActivity.java:147]: Here too

File: app/src/main/java/appricottsoftware/clarity/adapters/TabPagerAdapter.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+import appricottsoftware.clarity.fragments.BrowseFragment;
+import appricottsoftware.clarity.fragments.ChannelFragment;
+
+/**
+ * Created by karen on 1/19/18.
+ */
+
+public class TabPagerAdapter extends FragmentPagerAdapter {
+
+
```
Comment (lepatrick714) [inline app/src/main/java/appricottsoftware/clarity/adapters/TabPagerAdapter.java]: Get rid of these two lines. Not needed

File: app/src/main/java/appricottsoftware/clarity/adapters/TabPagerAdapter.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+                fragment = new ChannelFragment();
+                break;
+            case 1:
+                fragment = new BrowseFragment();
+                break;
+            default:
+                fragment = null;
+                break;
+        }
+
```
Comment (lepatrick714) [inline app/src/main/java/appricottsoftware/clarity/adapters/TabPagerAdapter.java]: Provide some small comments for the 3 cases

File: app/src/main/java/appricottsoftware/clarity/fragments/BrowseFragment.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+        View view = inflater.inflate(R.layout.fragment_browse, container, false);
+        ButterKnife.bind(this, view);
+        return view;
+    }
+
+    @Override
+    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
+
+        // Initialize view lookups, listeners
+    }
```
Comment (lepatrick714) [inline app/src/main/java/appricottsoftware/clarity/fragments/BrowseFragment.java]: Is this function done? If not, add a "TODO"

File: app/src/main/java/appricottsoftware/clarity/fragments/ChannelFragment.java [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+        View view = inflater.inflate(R.layout.fragment_channel, container, false);
+        ButterKnife.bind(this, view);
+        return view;
+    }
+
+    @Override
+    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
+
+        // Initialize view lookups, listeners
+    }
```
Comment (lepatrick714) [inline app/src/main/java/appricottsoftware/clarity/fragments/ChannelFragment.java]: Is this function done? If not, add a "TODO"

File: app/src/main/java/appricottsoftware/clarity/fragments/LikeFragment.java [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.fragment_like, container, false);
+        ButterKnife.bind(this, view);
+        return view;
+    }
+
+    @Override
+    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
+        // Initialize view lookups, listeners
+    }
```
Comment (lepatrick714) [inline app/src/main/java/appricottsoftware/clarity/fragments/LikeFragment.java:26]: Is this function done? If not, add a "TODO"

File: app/src/main/java/appricottsoftware/clarity/fragments/PlayerFragment.java [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.fragment_player, container, false);
+        ButterKnife.bind(this, view);
+        return view;
+    }
+
+    @Override
+    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
+        // Initialize view lookups, listeners
+    }
```
Comment (lepatrick714) [inline app/src/main/java/appricottsoftware/clarity/fragments/PlayerFragment.java:26]: Is this function done? If not, add a "TODO"

File: app/src/main/java/appricottsoftware/clarity/fragments/SettingFragment.java [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.fragment_setting, container, false);
+        ButterKnife.bind(this, view);
+        return view;
+    }
+
+    @Override
+    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
+        // Initialize view lookups, listeners
+    }
```
Comment (lepatrick714) [inline app/src/main/java/appricottsoftware/clarity/fragments/SettingFragment.java:26]: Is this function done? If not, add a "TODO"

**Review by lepatrick714 — CHANGES_REQUESTED** [review]

File: app/src/main/java/appricottsoftware/clarity/HomeActivity.java (hunk 1 above)
Comment (kkong006) [reply app/src/main/java/appricottsoftware/clarity/HomeActivity.java:91]: Indenting the 2nd line makes the code more readable. It is part of the previous line of code.

File: app/src/main/java/appricottsoftware/clarity/fragments/BrowseFragment.java (hunk 6 above)
Comment (kkong006) [reply app/src/main/java/appricottsoftware/clarity/fragments/BrowseFragment.java]: No it's not done. This is an app skeleton. All the files are "TODO".

File: app/src/main/java/appricottsoftware/clarity/adapters/TabPagerAdapter.java (hunk 5 above)
Comment (kkong006) [reply app/src/main/java/appricottsoftware/clarity/adapters/TabPagerAdapter.java]: Added explanation of which fragment is created for which tab. This is a standard FragmentPagerAdapter.

File: app/src/main/java/appricottsoftware/clarity/adapters/TabPagerAdapter.java (hunk 4 above)
Comment (kkong006) [reply app/src/main/java/appricottsoftware/clarity/adapters/TabPagerAdapter.java]: Done.

**Review by lepatrick714 — APPROVED** [review]

**Pull request merged**